### ISSUE TITLE
Rename folder

### ISSUE THREAD

**Issue by jmt85** [Qabel/qabel-android#372]
a created or uploaded Folder needs a renamin funktion

**Comment by wischweh** [comment]
How should name conflicts be handled?

As far as i recall it, we want to ignore special character and platform dependent issues but we still need to take care if there is two times the exact same name i.e. "TEST.png". Should be consistent with creating folders. 

But at the moment creating folders will just ignore the second operation **without any feedback** to the user. I think this should be considered a bug or area for enhancement.

**Comment by julianseeger** [comment]
>But at the moment creating folders will just ignore the second operation without any feedback to the user. I think this should be considered a bug or area for enhancement.

a user creating an already existing folder ... it's probably more intuitive when nothing happens (like now) because the user wanted a folder and he gets a folder ... everything fine. If you don't agree, you shoud create a separate issue

**Comment by wischweh** [comment]
@julianseeger i don't agree and will write an issue. Howeve with renaming it is different, because the expected operation was not executed. The user may have a folder with the expected name BUT different content.

**Comment by wischweh** [comment]
There are different options to handle this:
1. OK button is only enabled if the entered name is checked and ok (would need some more user feedback like in forms)
2. On Error the user could be asked if he wants to cancel the operation or overwrite the existing folder

**Comment by wischweh** [comment]
And another silent error (ignored by create folder) is entering an empty String. Again i advice against handling this silently. from my UX experience

**Comment by julianseeger** [comment]
jep, renaming is more critical, of course. Overwriting sounds evil to me (even without this option, the user can manually abort, delete the old folder and rename again).
So maybe
1.) +show little message like "name already in use"
3.) after the user confirms renaming, show error dialog with a string like in `1.)`

**Comment by jmt85** [comment]
ok, when the fodlername already exist, a littel messages is usefull. 
Overwriting without a dialog is not option. 
Overwriting with a confirm dialog is another issue and at the moment not important.
